### PR TITLE
feat: AlgoSearch Support

### PR DESCRIPTION
**Pull request by kevguy**
I'm making this PR as a part of **grow-algorand**, particularly [this issue](https://github.com/algorandfoundation/grow-algorand/issues/9).

Note that currently my changes to AlgoSearch is still in [PR](https://github.com/ori-shem-tov/algosearch/pull/1) and not merged yet, but it shouldn't affect the implementation of the support in the Sandbox here. Therefore, the newly configuration variables for AlgoSearch are using my Github repo URL and the associated branch for the time being (look at `config.dev`, `config.nightly` and `config.release`).

The following has been done
1. Added Docker build for AlgoSearch (`images/search`)
2. Updated `docker-compose.yml` to support CouchDB and AlgoSearch
3. Updated Sandbox script to
    - check health status of AlgoSearch in `status_helper`
    - check version of CouchDB in `version_helper`
    - let user enter into containers of AlgoSearch and CouchDB in `enter`
    - show the new arguments that Sandbox supports in `help`
    - test health status of AlgoSearch in `test`
4. AlgoSearch won't be available if no indexer is available in the sandbox
5. `./sandbox up` will try to open AlgoSearch in browser if it's available

To test it, simply run `./sandbox up` and go to `http://localhost:8000`.

### CODE REVIEW

File: config.testnet [hunk 1]
```diff
@@ -9,3 +9,6 @@ export INDEXER_URL=""
 export INDEXER_BRANCH=""
 export INDEXER_SHA=""
 export INDEXER_DISABLED="true"
+export ALGOSEARCH_URL="https://github.com/kevguy/algosearch"
```
Comment (winder) [inline config.testnet:12]: Does this tool work if no indexer is available? Perhaps leave the URL/Branch blank to disable algosearch as well.

**Review by winder — COMMENTED** [review]
This looks really cool! Could you add this to the end of the start function (if algosearch is installed):
```
      if [ $(uname) == "Darwin" ]; then
        open http://localhost:8000
      elif [ $(uname) == "Linux" ]; then
        xdg-open http://localhost:8000
      else
        echo "no browser detected..."
      fi
```

**Comment by winder** [comment]
Do you know why it says out of sync?
image

File: config.testnet (hunk 1 above)
Comment (kevguy) [reply config.testnet:12]: Just made a commit to disable the AlgoSearch if the indexer is not available in the sandbox. I was basically following what was done for indexer and made a mock NodeJS server serving status 400 if no `INDEXER_DISABLED` is `true`.

**Comment by kevguy** [comment]
> Do you know why it says out of sync?
> image

I looked at the source code and I think the value of if it's synced or nor is by default `false` and there's no code updating it. I'm still waiting to be in touch with the maintainer of the repo and see if he has any idea to fix it. Will keep you posted.

**Comment by kevguy** [comment]
@winder I'm also having trouble figuring our what went wrong with the CI build, because cliche as it is, the failed cases work on my machine.

**Comment by kevguy** [comment]
> This looks really cool! Could you add this to the end of the start function (if algosearch is installed):
> 
> ```
>       if [ $(uname) == "Darwin" ]; then
>         open http://localhost:8000
>       elif [ $(uname) == "Linux" ]; then
>         xdg-open http://localhost:8000
>       else
>         echo "no browser detected..."
>       fi
> ```

Just added the code in the new commit.

**Comment by ori-shem-tov** [comment]
> > Do you know why it says out of sync?
> > image
> 
> I looked at the source code and I think the value of if it's synced or nor is by default `false` and there's no code updating it. I'm still waiting to be in touch with the maintainer of the repo and see if he has any idea to fix it. Will keep you posted.

@kevguy I believe the issue is here: https://github.com/kevguy/algosearch/blob/181b4aab81a982059664d6975e619c596a3b4471/src/pages/Home/index.js#L30
you should use `response.data.ledger['current_round']` instead of `response.data.ledger.round`

File: docker-compose.yml [hunk 2]
```diff
@@ -52,3 +55,38 @@ services:
...
+      context: .
+      dockerfile: ./images/search/Dockerfile
+      args:
+        URL: "${ALGOSEARCH_URL}"
+        BRANCH: "${ALGOSEARCH_BRANCH}"
+        SHA: "${ALGOSEARCH_SHA}"
+    ports:
+      - 8000:8000
+    environment:
+      DISABLED: "${INDEXER_DISABLED}"
```
Comment (ori-shem-tov) [inline docker-compose.yml:71]: can you please use a different flag for AlgoSearch?

Comment (kevguy) [reply docker-compose.yml]: Done.

File: config.beta [hunk 3]
```diff
@@ -9,3 +9,7 @@ export INDEXER_URL="https://github.com/algorand/indexer"
 export INDEXER_BRANCH="develop"
 export INDEXER_SHA=""
 export INDEXER_DISABLED=""
+export ALGOSEARCH_URL="https://github.com/kevguy/algosearch"
```
Comment (ori-shem-tov) [inline config.beta:12]: @kevguy I've merged your PR in AlgoSearch repo, can you please update the `ALGOSEARCH_URL` and `ALGOSEARCH_BRANCH` in all the config files?

File: sandbox [hunk 4]
```diff
@@ -180,6 +186,8 @@ sandbox () {
     INDEXER_VERSION=$(dc exec indexer cmd/algorand-indexer/algorand-indexer -v) && echo ${INDEXER_VERSION} || curl -s "localhost:8980/health?pretty"
     statusline "Postgres version"
     dc exec indexer-db postgres --version
+    statusline "CouchDB version"
+    dc exec search-db curl http://localhost:5984
```
Comment (winder) [inline sandbox:190]: Does AlgoSearch have a version that can be included here?

**Comment by winder** [comment]
The broken CI looks related to the browser opening suggestion, I have two ideas for a fix.

1) Maybe opening in background will let the CI continue:
```
      if [ $(uname) == "Darwin" ]; then
        open http://localhost:8000 &
      elif [ $(uname) == "Linux" ]; then
        xdg-open http://localhost:8000 &
      else
        echo "no browser detected..."
      fi
```

2) special flag to disable:
.travis.yml:
```diff
-  - ./sandbox up $CONFIG
+  - SKIP_BROWSER=true ./sandbox up $CONFIG
```
```
      if [ ! -z $SKIP_BROWSER ]; then
        echo "skipping browser open..."
      elif [ $(uname) == "Darwin" ]; then
        open http://localhost:8000
      elif [ $(uname) == "Linux" ]; then
        xdg-open http://localhost:8000
      else
        echo "no browser detected..."
      fi
```

**Comment by kevguy** [comment]
Will work on it in a bit.

**Comment by kevguy** [comment]
@winder @ori-shem-tov I tested the code, which should be working with `./sandbox up -v` on all platforms. Would you mind doing some code review on this?

**Pull request closed without merging**